torch/_tensor.py

```diff
@@ -940,10 +940,14 @@ def to_sparse_csr(self):
                 while i < row_indices.size()[0] and row_indices[i] == irow:
                     i += 1
                 ro.append(i)
-
-            return torch.sparse_csr_tensor(torch.tensor(ro, dtype=row_indices.dtype),
-                                           coalesced_self.indices()[1], coalesced_self.values(),
-                                           size=coalesced_self.shape, dtype=coalesced_self.dtype)
+            device = coalesced_self.values().device
+            crow_indices = torch.tensor(ro, dtype=row_indices.dtype, device=device)
+            return torch.sparse_csr_tensor(crow_indices,
```

Comment:
Since we can trust COO tensor correctness, we should be able to produce inputs such that unsafe CSR constructor could be used. Let's address this in the context of https://github.com/pytorch/pytorch/issues/57381 rather than here.